### PR TITLE
feat: Orchestration, Rules relationship in rules response, New reference jsonld route

### PR DESCRIPTION
**Pull request by Visions-Nicolas**
- added a docker-compose.yml to launch a mongodb service and the app
- updated README.md
- added rules relationship with roles in response
- added a new parameters, populate, in /references/:referenceType to get the relation
- added a new routes : /references/referenceType/fileName to retrieve jsonld parsed
- updated tests and documentation

### CODE REVIEW

File: src/controllers/references.ts [hunk 1]
```diff
@@ -30,7 +31,33 @@ export const getReferencesByType = async (
 ) => {
   try {
     const { type } = req.params;
+    const { populate } = req.query;
     const data = await DefinedReference.find({ type });
+
+    if (type === "roles" && populate) {
```
Comment (FelixBole) [inline src/controllers/references.ts:37]: Verify populate as it should be a string in the query

File: scripts/dbUpdate.js [hunk 2]
```diff
@@ -107,6 +108,8 @@ mongoose
             ? jsonld["name"]
             : "";
 
+          const uid = jsonld["uid"] ?? fileName.slice(0,-5);
```
Comment (FelixBole) [inline scripts/dbUpdate.js:111]: Rule filename === uid

File: src/controllers/references.ts [hunk 3]
```diff
@@ -30,7 +31,33 @@ export const getReferencesByType = async (
...
+      await Promise.all(
+        data.map(async (role) => {
+          const roleJsonldData = JSON.parse(role.jsonld);
+          const test = await DefinedReference.find(
```
Comment (FelixBole) [inline src/controllers/references.ts:41]: Change test variable name.
Might also be interesting to .lean() the DefinedReference.find() to get the object directly

**Review by FelixBole — CHANGES_REQUESTED** [review]
Slight changes requested but nothing too serious

File: src/controllers/references.ts (hunk 1 above)
Comment (Visions-Nicolas) [reply src/controllers/references.ts]: populate verified if equal "1" or "true"

File: scripts/dbUpdate.js (hunk 2 above)
Comment (Visions-Nicolas) [reply scripts/dbUpdate.js]: replace with filename

File: src/controllers/references.ts (hunk 3 above)
Comment (Visions-Nicolas) [reply src/controllers/references.ts]: added lean() and renamed variable

**Pull request merged**